helm.el

```diff
@@ -3114,12 +3119,35 @@ and `helm-pattern'."
              (if (string-match "[[:upper:]]" pattern) nil t)))
     (t helm-case-fold-search)))
 
-(defun helm-match-from-candidates (cands matchfns match-part-fn limit source)
-  (let (matches)
+
+(defun helm--mapconcat-initials-pattern (pattern seperators)
```

Comment:
No, it works best when used together with `helm-flx`. This keeps the number of matches sent to `flx` small-ish, but tries harder to have the best candidates available among those sent to `flx`.

You could use the latest iteration of the regex to replace helm's stock one for fuzzy matching,
and even without flx it should work quite well, but I haven't tried it.